### PR TITLE
Fix `LabelMapAxis` so it doesn't reorder the labels

### LINKED ISSUE #5384: LabelMapAxis re-orders labels
**Gammapy version**
v1.1

**Bug description**
When instancing a `LabelMapAxis` the labels are re-ordered / sorted. This is problematic because it should keep the order, that corresponds to the  data.

**Expected behavior**
Do not re-order / sort the labels.

**To Reproduce**
```python
import numpy as np
from gammapy.maps import LabelMapAxis

labels = np.array([3, 2, 1, 4, 5])
print(labels)
print(LabelMapAxis(labels).center)
```
Which prints:
```
[3 2 1 4 5]
[1 2 3 4 5]
```

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
Resolves #5384

Note: had to ensure that labels was an array type, as this is what np.unique would do

### CODE REVIEW

File: gammapy/maps/tests/test_axes.py [hunk 1]
```diff
@@ -793,6 +793,17 @@ def test_label_map_axis_basics():
     axis_copy = axis.copy()
     assert axis_copy.name == "label-axis"
 
+    # Ensure order is correct
+    labels = np.array([1, 4, 5])
```
Comment (registerrier) [inline gammapy/maps/tests/test_axes.py:797]: Could you add a test checking that alphabetical order of labels is not imposed?

**Review by registerrier — APPROVED** [review]
Thanks @Astro-Kirsty . Looks good. It just misses one test for non re-ordering.

**Review by adonath — APPROVED** [review]
Thanks @Astro-Kirsty, I have no further comments.

**Review by QRemy — APPROVED** [review]
Thanks @Astro-Kirsty merging now

**Pull request merged**